File: application/vivid_estate_frontend_flutter/lib/Classes/InputValidator.dart
// This Class Will Validate Different Types of Input from the User
class InputValidator {
  // The Pattern of our email to be validate with
  final emailPattern =
      r'^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';

  // Parameters to Validate Our Password
  final passwordLength = 8;

  // Store the Error Message during the Validation
  var message = "";

  // Validate the Email Address of the User
  bool validateEmailAddress(String email) {
    // Check if Password is empty
    if (email.isEmpty) {
      message = "Please provide a email address";
      return false;
    }

    // Create a Regex Object
    RegExp regex = RegExp(emailPattern);

    // Perform Validation and return status
    var result = regex.hasMatch(email);

    // Check if email is valid or not
    if (result == false) {
      message = "Enter a valid email address";
    }

    // Send our result back
    return result;
  }

  // Make Sure Password is Strong
  bool validatePassword(String password) {
    // Check if Password is empty
    if (password.isEmpty) {
      message = "Please provide a password";
      return false;
    }

    // Length Requirement
    if (password.length < passwordLength) {
      message = "Password must be of atleast 8 characters";
      return false;
    }

    // Capital Letter Requirement
    bool hasCapital = false;
    for (int i = 0; i < password.length; i++) {
      if (password.codeUnitAt(i) >= 65 && password.codeUnitAt(i) <= 90) {
        hasCapital = true;
        break;
      }
    }

    // Check if Has Capital Letter or not
    if (hasCapital == false) {
      message = "Password must have atleast 1 capital letter";
    }

    return hasCapital;
  }
}

File: server/backendApplication/Components/InputValidator.py
import re

class InputValidator:

    # Validate The Email Address Of the User
    def validateEmailAddress(self, email:str):

        # Pattern For Email Address
        email_pattern = re.compile( r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)")
        
        # Match the email against the regex pattern
        if re.match(email_pattern, email):

            # The Pattern is match
            return True
        else:

            # Pattern Not Able to Match
            return False

    # Validate the Password of the User
    def validatePassword(self, password):
        
        # Check if Password is empty
        if not password:
            return False

        # Length Requirement
        if len(password) < 8:
            return False

        # Capital Letter Requirement
        has_capital = any(char.isupper() for char in password)

        # Check if Has Capital Letter or not
        if not has_capital:
            return False

        return True
